classes/04class/exercises/c04-iac02/fabioacp/variables.tf

```diff
@@ -0,0 +1,145 @@
+variable "vpc_cidr" {
+  description = "VPC network"
+  default     = "10.0.0.0/16"
+}
+
+variable "instance_tenancy" {
+  type        = string
+  description = "defult or dedicated"
+  default     = "default"
+}
+
+variable "dns_support" {
+  type    = bool
+  default = true
+}
+
+variable "dns_host_names" {
+  type    = bool
+  default = true
+}
+
+variable "name" {
+  description = "Name to be used on all the resources as identifier"
+  type        = string
+  default     = ""
+}
+
+variable "azs" {
+  description = "A list of availability zones names or ids in the region"
+  type        = list(string)
+  default     = []
+}
+
+variable "private_subnet_names" {
+  description = "A list of private subnet names"
+  type        = list(string)
+  default     = []
+}
+
+variable "public_subnet_names" {
+  description = "A list of public subnet names"
+  type        = list(string)
+  default     = []
+}
+
+variable "private_subnets" {
+  description = "A list of private subnets inside the VPC"
+  type        = list(string)
+  default     = []
+}
+
+variable "public_subnets" {
+  description = "A list of public subnets inside the VPC"
+  type        = list(string)
+  default     = []
+}
+
+variable "public_subnet_suffix" {
+  description = "Suffix to append to public subnets name"
+  type        = string
+  default     = "public"
+}
+
+variable "private_subnet_suffix" {
+  description = "Suffix to append to private subnets name"
```

Comment:
another option is grouping these variables in a `map(object())` data structure too, so you'd have a list of KV pairs each one containing objects you defined as `name`, `cidr`, `is_public`, etc..